src/main/java/javax/validation/ConstraintViolationException.java

```diff
@@ -32,7 +34,7 @@ public ConstraintViolationException(String message,
 			this.constraintViolations = null;
 		}
 		else {
-			this.constraintViolations = new HashSet<ConstraintViolation<?>>( constraintViolations );
+			this.constraintViolations = new HashSet<>( constraintViolations );
```

Comment:
Agreed it's odd, and I also cannot see a case where I'd do it. But then I don't see a good reason to rule it out. The one example I can see is that you catch the exception, add another constraint violation with some additional info and re-throw it. It's not a usage I'd encourage, but think it's alright to allow it.